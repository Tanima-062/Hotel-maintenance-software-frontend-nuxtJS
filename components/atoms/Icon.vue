<template>
  <div :class="styles">
    <svg
      class="w-7 h-7 fill-current"
      :class="`${color}`"
      :style="`width: ${iconWidth}px; height: ${iconHeight}px`"
      :viewBox="`0 0 24 24`"
    >
      <path :d="icon" />
    </svg>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  icon!: string

  @Prop({ default: 'text-white' })
  color!: string

  @Prop({ default: '' })
  bgColor!: string

  @Prop({ default: false })
  round!: boolean

  @Prop({ default: 24 })
  iconWidth!: number

  @Prop({ default: 24 })
  iconHeight!: number

  @Prop({ default: '' })
  extraClass!: string

  get styles(): string {
    const classList = []
    if (this.bgColor) {
      classList.push(`${this.bgColor}`)
    }
    if (this.round) {
      classList.push('rounded-full')
    }
    if (this.extraClass) {
      classList.push(this.extraClass)
    }
    return classList.join(' ')
  }
}
</script>
