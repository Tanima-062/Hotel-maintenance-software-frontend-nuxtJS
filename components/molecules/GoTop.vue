<template>
  <div
    class="opacity-80 w-24 rounded-bl-lg rounded-br-md px-3 py-1 bg-black-light cursor-pointer"
    :class="extraClass"
    @click="scrollToTopWindow"
  >
    <div class="flex flex-row items-center justify-center">
      <Icon :icon="topIcon" color="text-white" extra-class="-ml-2 mr-1" />
      <p class="text-xs text-white">To top</p>
    </div>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'
import { mdiChevronUp } from '@mdi/js'

@Component({
  components: {
    Icon: () => import('~/components/atoms/Icon.vue'),
  },
})
export default class extends Vue {
  @Prop({ default: '' })
  extraClass!: string

  @Prop({ required: true })
  toTopId!: string

  private topIcon: string = mdiChevronUp
  private scrollToTopWindow(): void {
    const scroll = document.getElementById(this.toTopId)
    if (!scroll) return
    scroll.scrollTop = 0
  }
}
</script>
