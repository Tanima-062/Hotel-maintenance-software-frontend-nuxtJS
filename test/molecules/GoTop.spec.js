import { mount } from '@vue/test-utils'
import GoTop from '@/components/molecules/GoTop.vue'
import Icon from '@/components/atoms/Icon.vue'

describe('Molecules GoTop Check', () => {
  test('GoTop Test', () => {
    const props = {
      toTopId: 'testTopId',
    }
    const wrapper = mount(GoTop, {
      propsData: props,
      stubs: {
        Icon,
      },
    })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
