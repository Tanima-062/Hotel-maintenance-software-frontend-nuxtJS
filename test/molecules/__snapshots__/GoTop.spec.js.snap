// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules GoTop Check GoTop Test 1`] = `
<div
  class="opacity-80 w-24 rounded-bl-lg rounded-br-md px-3 py-1 bg-black-light cursor-pointer"
>
  <div
    class="flex flex-row items-center justify-center"
  >
    <div
      class="-ml-2 mr-1"
    >
      <svg
        class="w-7 h-7 fill-current text-white"
        style="width: 24px; height: 24px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
        />
      </svg>
    </div>
     
    <p
      class="text-xs text-white"
    >
      To top
    </p>
  </div>
</div>
`;
